File: lyedu-api/src/main/java/com/lyedu/entity/Video.java
package com.lyedu.entity;

import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 视频实体
 *
 * @author LyEdu Team
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class Video extends BaseEntity {

    /**
     * 课程ID
     */
    private Long courseId;

    /**
     * 章节ID
     */
    private Long chapterId;

    /**
     * 课程名称（展示用，非表字段）
     */
    private String courseName;

    /**
     * 章节名称（展示用，非表字段）
     */
    private String chapterName;

    /**
     * 视频标题
     */
    private String title;

    /**
     * 视频地址
     */
    private String url;

    /**
     * 视频时长（秒）
     */
    private Integer duration;

    /**
     * 排序
     */
    private Integer sort;
}

File: lyedu-admin/src/api/video.ts
import request from '@/utils/request'

export interface Video {
  id: number
  courseId: number
  chapterId?: number
  courseName?: string
  chapterName?: string
  title: string
  url: string
  cover?: string
  duration?: number
  sort: number
  playCount?: number
  likeCount?: number
}

export interface PageResult<T> {
  records: T[]
  total: number
  current: number
  size: number
  pages: number
}

export const getVideoPage = (params: {
  page: number
  size: number
  courseId?: number
  keyword?: string
}) => {
  return request.get<PageResult<Video>>('/video/page', { params })
}

export const getVideoById = (id: number) => {
  return request.get<Video>(`/video/${id}`)
}

export const createVideo = (data: Partial<Video>) => {
  return request.post('/video', data)
}

export const updateVideo = (id: number, data: Partial<Video>) => {
  return request.put(`/video/${id}`, data)
}

export const deleteVideo = (id: number) => {
  return request.delete(`/video/${id}`)
}

export const getVideosByCourseId = (courseId: number) => {
  return request.get<Video[]>(`/video/course/${courseId}`)
}

export const getVideosByChapterId = (chapterId: number) => {
  return request.get<Video[]>(`/video/chapter/${chapterId}`)
}

